### PR TITLE
functions for doing stream writes to disk and s3

### PR DESCRIPTION
**Pull request by maxkozlovsky**
add traits and implementations for accepting stream of Bytes for writes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -90,6 +90,7 @@ sqlx = { version = "0.7.3", features = [
     "json",
 ] }
 strum = { version = "0.26.1", features = ["derive"] }
+tempfile = "3.2.0"
```
Comment (diptanu) [inline Cargo.toml:93]: Add this to [dev-dependencies] since you are using this only in tests.

File: src/blob_storage/disk.rs [hunk 2]
```diff
@@ -28,6 +28,24 @@ impl BlobStorageWriter for DiskStorage {
...
+        Ok(path)
+    }
+
+    async fn put_stream(
+        &self,
+        key: &str,
+        data: BoxStream<'static, Result<Bytes>>,
+    ) -> Result<String, anyhow::Error> {
+        let path = format!("{}/{}", self.config.path, key);
+        let mut file = File::create(&path).await?;
```
Comment (diptanu) [inline src/blob_storage/disk.rs:42]: Should we use temp files here and then move the file if it's successful?

Comment (maxkozlovsky) [reply src/blob_storage/disk.rs:42]: In principle yes, I was following the original implementation. It also has the problem if the write fails/partially succeeds. I can add a todo for now.

File: Cargo.toml (hunk 1 above)
Comment (maxkozlovsky) [reply Cargo.toml]: ok

**Pull request closed without merging**